ui/src/app/streams/stream-deploy/stream-deploy.component.ts

```diff
@@ -21,66 +27,352 @@ import { takeUntil } from 'rxjs/operators';
+      appsVersion.addControl(app.name, control);
+    });
+
+    // Useful methods for FormArray
+    const add = (array: FormArray) => {
+      const group = new FormGroup({
+        'property': new FormControl('', [StreamDeployValidator.key]),
+        'global': new FormControl('')
+      });
+      config.apps.forEach((app) => {
+        group.addControl(app.name, new FormControl(''));
+      });
+      array.push(group);
+    };
+    const isEmpty = (dictionary): boolean => Object.entries(dictionary).every((a) => a[1] === '');
+    const clean = (val: Array<any>, array: FormArray) => {
+      const toRemove = val.map((a, index) =>
+        ((index < (val.length - 1) && isEmpty(a)) ? index : null)).filter((a) => a != null);
+
+      toRemove.reverse().forEach((a) => {
+        array.removeAt(a);
+      });
+      if (!isEmpty(val[val.length - 1])) {
+        return add(array);
+      }
+    };
+
+    // Dynamic App properties
+    const globalControls: FormArray = new FormArray([]);
+    add(globalControls);
+    globalControls.valueChanges.subscribe((val: Array<any>) => {
+      clean(val, globalControls);
+    });
+
+    // Dynamic Platform properties
+    const specificPlatformControls: FormArray = new FormArray([]);
+    add(specificPlatformControls);
+    specificPlatformControls.valueChanges.subscribe((val: Array<any>) => {
+      clean(val, specificPlatformControls);
+    });
+
+    groups.addControl('deployers', deployers);
+    groups.addControl('appsVersion', appsVersion);
+    groups.addControl('global', globalControls);
+    groups.addControl('specificPlatform', specificPlatformControls);
+
+    // Free Text
+    const fileGroup = new FormGroup({
+      file: new FormControl(),
+      properties: new FormControl('', StreamDeployValidator.properties)
+    });
+
+    return {
+      builder: groups,
+      builderAppsProperties: builderAppsProperties,
+      file: fileGroup,
+      config: config
+    };
+  }
+
+  /**
+   * Get properties raw
+   *
+   * @param value
+   */
+  toRaw(value: {
+    builder: FormGroup, builderAppsProperties: {}, file: FormGroup,
+    config: StreamDeployConfig
+  }): Array<{ key: string, value: string, status: string }> {
+    const result = [];
+    const isEmpty = (control: AbstractControl) => !control || (control.value === '' || control.value === null);
+    const status = (controls: AbstractControl[]): string =>
+      controls.every((a) => !a.invalid) ? 'valid' : 'invalid';
+
+    if (this.state.view === 'builder') {
+      const deployers: FormArray = value.builder.get('deployers') as FormArray;
+      const appsVersion: FormGroup = value.builder.get('appsVersion') as FormGroup;
+      const global: FormArray = value.builder.get('global') as FormArray;
+      const specificPlatform: FormArray = value.builder.get('specificPlatform') as FormArray;
+
+      // Platform
+      if (!isEmpty(value.builder.get('platform'))) {
+        result.push({
+          key: 'deployer.platform',
```

Comment:
The key `deployer.platform` looks incorrect as it needs to be `spring.cloud.dataflow.skipper.platformName`